src/test/java/pl/creative/rental_server/core/features/moderator/category/CategoryServiceTest.java

```diff
@@ -0,0 +1,131 @@
+import pl.creative.rental_server.core.features.moderator.item.dto.ItemMapper;
+import pl.creative.rental_server.core.global.exception.notFound.CategoryException;
+import pl.creative.rental_server.core.global.exception.notFound.CategoryNotFound;
+import pl.creative.rental_server.core.global.handlersAndUtils.RandomIdHandler;
+import pl.creative.rental_server.db.entities.Category;
+import pl.creative.rental_server.db.repository.CategoryRepository;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceTest {
+
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private CategoryMapper categoryMapper;
+    @Mock
+    private RandomIdHandler randomIdHandler;
+    @Mock
+    private ItemMapper itemMapper;
+    private CategoryService underTest;
+
+    @BeforeEach
+    void setUp() {
+        underTest = new CategoryService(categoryMapper, categoryRepository, randomIdHandler, itemMapper);
+    }
+
+    @Test
+    void addCategory_shouldAddGivenCategory() {
+        // given
+        FillCategoryDto fillCategoryDto = new FillCategoryDto("categoryName", "description");
+        Category category = new Category("categoryId", "categoryName", "description");
+        given(categoryRepository.findCategoryByCategoryName(anyString())).willReturn(Optional.empty());
+        given(categoryMapper.mapFillCategoryDtoToCategory(fillCategoryDto)).willReturn(category);
+        // when
+        underTest.addCategory(fillCategoryDto);
+        // then
+        ArgumentCaptor<Category> studentArgumentCaptor = ArgumentCaptor.forClass(Category.class);
+        verify(categoryRepository).save(studentArgumentCaptor.capture());
+        Category capturedStudent = studentArgumentCaptor.getValue();
+        assertThat(capturedStudent.getClass()).isEqualTo(Category.class);
+        assertThat(capturedStudent).isEqualTo(category);
+        assertThat(capturedStudent.equals(category)).isTrue();
+    }
+
+    @Test
+    void addCategory_shouldThrowCategoryException_shouldNotAddGivenCategory() {
+        // given
+        String categoryName = "categoryName";
+        FillCategoryDto fillCategoryDto = new FillCategoryDto(categoryName, "description");
+        Category category = new Category("categoryId", categoryName, "description");
+        given(categoryRepository.findCategoryByCategoryName(anyString())).willReturn(Optional.of(category)); //the category already exists
+//        given(categoryMapper.mapFillCategoryDtoToCategory(fillCategoryDto)).willReturn(category); //this line is unnecessary!
```

Comment:
Clean up code ( do not leave unnessesary comments when we are not making 100% nessesary things )